lint-workflow/lint.py

```diff
@@ -292,9 +295,17 @@ def lint(filename):
                             logging.info("Skipping local action in workflow.")
                             break
 
-                        # If the step has a 'uses' key, check value hash, except bitwarden actions.
+                        # If the step has a 'uses' key, check if actions are in supported actions list and also value hash, except bitwarden actions.
                         if "bitwarden/gh-actions" not in path:
                             try:
+                                # Check if actions are in supported actions list.
+                                if path != supported_actions[path]:
```

Comment:
What happens if a key is not in a python dictionary?

Reply:
Then we are not supporting the action, hence, it will send the warning. The warning will be replaced by errors, when we are failing the lint.

Reply:
What I have now is checking for the unavailability of the action, if true, then throw the warning.

Reply:
The intended use makes sense. Thanks!

I was wondering specifically about the python code if a `key` doesn't exist in a `dict`. For instance:

```python
a = {
  'a': 1,
  'b': 2,
  'c': 3
}

print(a['d'])
print('got here')
```

The expected output here would be a `KeyError`. In the previous `dict` implementation of the linter, the `if` statement would have errored and exited the program instead of entering the `if` block.

Reply:
Oh, that's an oversight  🫥🫥! Thank youuuu ❤️